lib/rubocop/cop/style/unless_logical_operators.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for the use of logical operators in an unless condition.
+      #
+      # @example EnforcedStyle: no_mixed_logical_operator (default)
+      #   # bad
+      #   return unless a || b && c
+      #   return unless a && b || c
+      #
+      #   # good
+      #   return unless a || b || c
+      #   return unless a && b && c
+      #   return unless a?
+      #
+      # @example EnforcedStyle: no_logical_operator
+      #   # bad
+      #   return unless a || b && c
+      #   return unless a && b || c
+      #   return unless a || b || c
+      #   return unless a && b && c
+      #
+      #   # good
+      #   return unless a?
+      class UnlessLogicalOperators < Base
+        include ConfigurableEnforcedStyle
+
+        NO_MIXED_LOGICAL_OPERATOR_MSG = 'Do not use both && and || in an unless condition.'
```

Comment:
It's best to put the operators in backticks, so they'd be coloured in the terminal output.